### PR TITLE
Update .pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by ro4i7**
Solving issue #275 
Please check @kantord , i have used yamllint
i have done it under hacktoberfest2023

### CODE REVIEW

**Comment by kantord** [comment]
Hi, looks like the lint tasks are still failing

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,51 +1,55 @@
...
+
+  - repo: https://github.com/dhruvmanila/remove-print-statements
+    rev: v0.5.2
+    hooks:
+      - id: remove-print-statements
+        args:
+          - --verbose
+          - --ignore=tests/conftest.py
+
+  - repo: https://github.com/adrienverge/yamllint
```
Comment (kantord) [inline .pre-commit-config.yaml:52]: Keep in mind we already have this hook:  https://github.com/jumanjihouse/pre-commit-hook-yamlfmt

could it cause a problem? I mean, will the 2 diffferent yaml linters/formatters interfere with each other? If both are actually needed, then could it be done in such a way that the second one will *only* care about the line length (by ignoring all other rules/formatting?)

Comment (ro4i7) [reply .pre-commit-config.yaml:54]: yes u r right, I have configured one of the linter to ignore line length, by making `yamlfmt` to only check line length

**Comment by stroncod** [comment]
Per docs in yamlfmt:

> yamlfmt only works with valid YAML files, so I recommend to use yamllint and yamlfmt together.

So yamllint won't affect the behavior. Then to run yamlfmt to only check line-width is with `--width` that's why there is an error in the lint check. Default is 150 so technically is already doing formatting for YAML files, do you want that lower? It seems that's the only modification is need. @ro4i7 go ahead an finish the PR for your Hacktoberfest.

**Comment by ro4i7** [comment]
> Per docs in yamlfmt:
> 
> > yamlfmt only works with valid YAML files, so I recommend to use yamllint and yamlfmt together.
> 
> So yamllint won't affect the behavior. Then to run yamlfmt to only check line-width is with `--width` that's why there is an error in the lint check. Default is 150 so technically is already doing formatting for YAML files, do you want that lower? It seems that's the only modification is need. @ro4i7 go ahead an finish the PR for your Hacktoberfest.

@stroncod ,i  have set the width. please check.

**Comment by kantord** [comment]
let me see

**Review by kantord — CHANGES_REQUESTED** [review]
Now it's giving these errors


```
usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
               [-p] [--implicit_start | --explicit_start]
               [--implicit_end | --explicit_end] [-n]
               [FILE_NAME ...]
yamlfmt: error: unrecognized arguments: --line-length=80
usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
               [-p] [--implicit_start | --explicit_start]
               [--implicit_end | --explicit_end] [-n]
               [FILE_NAME ...]
yamlfmt: error: unrecognized arguments: --line-length=80

remove print statements..................................................Passed
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 2

usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
                [-f {parsable,standard,colored,github,auto}] [-s]
                [--no-warnings] [-v]
                [FILE_OR_DIR ...]
yamllint: error: unrecognized arguments: --width=[80](https://github.com/kantord/SeaGOAT/actions/runs/6388169149/job/17344055464?pr=286#step:6:81)
usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
                [-f {parsable,standard,colored,github,auto}] [-s]
                [--no-warnings] [-v]
                [FILE_OR_DIR ...]
yamllint: error: unrecognized arguments: --width=80

```

btw you can run it locally using `pre-commit run --all-files` that should also make sure that it passes on the CI

**Comment by ro4i7** [comment]
> Now it's giving these errors
> 
> ```
> usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
>                [-p] [--implicit_start | --explicit_start]
>                [--implicit_end | --explicit_end] [-n]
>                [FILE_NAME ...]
> yamlfmt: error: unrecognized arguments: --line-length=80
> usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
>                [-p] [--implicit_start | --explicit_start]
>                [--implicit_end | --explicit_end] [-n]
>                [FILE_NAME ...]
> yamlfmt: error: unrecognized arguments: --line-length=80
> 
> remove print statements..................................................Passed
> yamllint.................................................................Failed
> - hook id: yamllint
> - exit code: 2
> 
> usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
>                 [-f {parsable,standard,colored,github,auto}] [-s]
>                 [--no-warnings] [-v]
>                 [FILE_OR_DIR ...]
> yamllint: error: unrecognized arguments: --width=[80](https://github.com/kantord/SeaGOAT/actions/runs/6388169149/job/17344055464?pr=286#step:6:81)
> usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
>                 [-f {parsable,standard,colored,github,auto}] [-s]
>                 [--no-warnings] [-v]
>                 [FILE_OR_DIR ...]
> yamllint: error: unrecognized arguments: --width=80
> ```
> 
> btw you can run it locally using `pre-commit run --all-files` that should also make sure that it passes on the CI

it used --width 80 format, in both yamlint and yamlfmt,

**Comment by ro4i7** [comment]
> > Now it's giving these errors
> > ```
> > usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
> >                [-p] [--implicit_start | --explicit_start]
> >                [--implicit_end | --explicit_end] [-n]
> >                [FILE_NAME ...]
> > yamlfmt: error: unrecognized arguments: --line-length=80
> > usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
> >                [-p] [--implicit_start | --explicit_start]
> >                [--implicit_end | --explicit_end] [-n]
> >                [FILE_NAME ...]
> > yamlfmt: error: unrecognized arguments: --line-length=80
> > 
> > remove print statements..................................................Passed
> > yamllint.................................................................Failed
> > - hook id: yamllint
> > - exit code: 2
> > 
> > usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
> >                 [-f {parsable,standard,colored,github,auto}] [-s]
> >                 [--no-warnings] [-v]
> >                 [FILE_OR_DIR ...]
> > yamllint: error: unrecognized arguments: --width=[80](https://github.com/kantord/SeaGOAT/actions/runs/6388169149/job/17344055464?pr=286#step:6:81)
> > usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
> >                 [-f {parsable,standard,colored,github,auto}] [-s]
> >                 [--no-warnings] [-v]
> >                 [FILE_OR_DIR ...]
> > yamllint: error: unrecognized arguments: --width=80
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > btw you can run it locally using `pre-commit run --all-files` that should also make sure that it passes on the CI
> 
> it used --width 80 format, in both yamlint and yamlfmt,

it used --width format , but still it giving linting error in yamlint

**Comment by stroncod** [comment]
That's because 

> > > Now it's giving these errors
> > > ```
> > > usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
> > >                [-p] [--implicit_start | --explicit_start]
> > >                [--implicit_end | --explicit_end] [-n]
> > >                [FILE_NAME ...]
> > > yamlfmt: error: unrecognized arguments: --line-length=80
> > > usage: yamlfmt [-h] [-m MAPPING] [-s SEQUENCE] [-o OFFSET] [-c] [-w WIDTH]
> > >                [-p] [--implicit_start | --explicit_start]
> > >                [--implicit_end | --explicit_end] [-n]
> > >                [FILE_NAME ...]
> > > yamlfmt: error: unrecognized arguments: --line-length=80
> > > 
> > > remove print statements..................................................Passed
> > > yamllint.................................................................Failed
> > > - hook id: yamllint
> > > - exit code: 2
> > > 
> > > usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
> > >                 [-f {parsable,standard,colored,github,auto}] [-s]
> > >                 [--no-warnings] [-v]
> > >                 [FILE_OR_DIR ...]
> > > yamllint: error: unrecognized arguments: --width=[80](https://github.com/kantord/SeaGOAT/actions/runs/6388169149/job/17344055464?pr=286#step:6:81)
> > > usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA]
> > >                 [-f {parsable,standard,colored,github,auto}] [-s]
> > >                 [--no-warnings] [-v]
> > >                 [FILE_OR_DIR ...]
> > > yamllint: error: unrecognized arguments: --width=80
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > btw you can run it locally using `pre-commit run --all-files` that should also make sure that it passes on the CI
> > 
> > 
> > it used --width 80 format, in both yamlint and yamlfmt,
> 
> it used --width format , but still it giving linting error in yamlint

Try using the flag on yamlfmt and not yamlint as it does not have a flag for that.

**Comment by ro4i7** [comment]
> Try using the flag on yamlfmt and not on yamlint.

ohk i have done it, please check

**Comment by kantord** [comment]
> > Try using the flag on yamlfmt and not on yamlint.
> 
> ohk i have done it, please check

hi, it's still failing from what I can see

**Comment by ro4i7** [comment]
> > > Try using the flag on yamlfmt and not on yamlint.
> > 
> > 
> > ohk i have done it, please check
> 
> hi, it's still failing from what I can see

Hello, Yes it's still failing

**Comment by stroncod** [comment]
> > > > Try using the flag on yamlfmt and not on yamlint.
> > > 
> > > 
> > > ohk i have done it, please check
> > 
> > 
> > hi, it's still failing from what I can see
> 
> Hello, Yes it's still failing. Any suggestions?

It fails because yamlfmt does not modify the files, only notifies you. You can run a script to solve that after yamlfmt tells you which files you need

**Comment by kantord** [comment]
Also now the branch needs to be rebased it seems

**Pull request closed without merging**